### PR TITLE
Add support for setting device time

### PR DESCRIPTION
**Pull request by anthonypdawson**
Implementing setting device time for the Android platform across this repo, appium-base-driver and python-client

### CODE REVIEW

File: lib/commands/general.js [hunk 1]
```diff
@@ -39,6 +39,16 @@ commands.getDeviceTime = async function () {
   }
 };
 
+commands.setDeviceTime = async function (dateTime) {
```
Comment (mykola-mokhnach) [inline lib/commands/general.js]: Please add proper docstring

File: lib/commands/general.js [hunk 2]
```diff
@@ -39,6 +39,16 @@ commands.getDeviceTime = async function () {
...
+  log.info('Attempting to set device date and time');
+  try {
+    let out = await this.adb.shell(['date', dateTime]);
+    return out.trim();
+  } catch (err) {
+    log.errorAndThrow(`Could not set device date and time: ${err}`);
```
Comment (mykola-mokhnach) [inline lib/commands/general.js]: err.message

File: lib/commands/general.js [hunk 3]
```diff
@@ -39,6 +39,16 @@ commands.getDeviceTime = async function () {
   }
 };
 
+commands.setDeviceTime = async function (dateTime) {
+  log.info('Attempting to set device date and time');
```
Comment (mykola-mokhnach) [inline lib/commands/general.js]: I assume a better place for such stuff would be appium-adb module. WDYT @imurchie ?

Comment (anthonypdawson) [reply lib/commands/general.js]: I don't mind making that change, these are my first PRs.  I was basically working alongside getDeviceTime

Comment (imurchie) [reply lib/commands/general.js]: Yes, this ought to be done in the `appium-adb` module. We should not really ever be calling `adb.shell` outside of there, if possible (I know we break that all over the place, but shouldn't :( ).

**Comment by mykola-mokhnach** [comment]
So, as a conclusion:
- The time setting functionality should be moved to appium-adb module
- The command itself should be available via `mobile:` interface or as a capability value

**Pull request closed without merging**